### PR TITLE
chore(cli): update package.json repository config

### PR DESCRIPTION
**Pull request by tw15egan**
Noticed after https://github.com/carbon-design-system/carbon/commit/05649fbc7b920b4df3888a47b17dd29f9fb84a2b that the changes were overwritten by our sync tool. This changes the CLI tool to add the correct monorepo repository structure so relative links will work on NPM

#### Changelog

**Changed**

- Updates the CLI tool to render the `repository` field as an object so monorepo relative paths work correctly

**Removed**

- unneeded variable

#### Testing / Reviewing

@joshblack wasn't sure how to test this, let me know if there is something I can run to verify 👍

### CODE REVIEW

File: packages/cli/src/commands/sync/package.js [hunk 1]
```diff
@@ -73,7 +70,11 @@ function run({ packagePaths }) {
   return Promise.all(
     packagePaths.map(
       async ({ packageJsonPath, packageJson, packageFolder }) => {
-        packageJson.repository = `${REPO_URL_BASE}/${packageFolder}`;
+        packageJson.repository = {
+          type: 'git',
+          url: 'https://github.com/carbon-design-system/carbon.git',
+          directory: `${packageFolder}`,
```
Comment (joshblack) [inline packages/cli/src/commands/sync/package.js]:
```suggestion
          directory: packageFolder,
```

**Review by joshblack — APPROVED** [review]
Looks great! Just had one style comment but seems perfect regardless

**Comment by joshblack** [comment]
@tw15egan to test it locally you could run at the top-level:

```bash
yarn carbon-cli sync package
```

And it should work! Also totally feel free to commit those changes too if you want.

**Review by emyarod — APPROVED** [review]
looks good to me

**Pull request merged**